openbas-front/src/components/ChainedTimeline.tsx

```diff
@@ -148,33 +150,102 @@ const ChainedTimelineFlow: FunctionComponent<Props> = ({
     return Math.round(((position.x) / (gapSize / minutesPerGapAllowed[minutesPerGapIndex])) * 60);
   };
 
+  /**
+   * Move item from an index to another one
+   * @param array the array to update
+   * @param to the target index
+   * @param from the origin index
+   */
+  const moveItem = (array: NodeInject[], to: number, from: number) => {
+    const item = array[from];
+    array.splice(from, 1);
+    array.splice(to, 0, item);
+    return array;
+  };
+
+  /**
+   * Calculate a bounding box for an index
+   * @param currentNode the node to calculate the bounding box for
+   * @param nodesAvailable the nodes
+   */
+  const calculateBoundingBox = (currentNode: NodeInject, nodesAvailable: NodeInject[]) => {
+    if (currentNode.data.inject?.inject_depends_on) {
+      const nodesId = Object.keys(currentNode.data.inject?.inject_depends_on);
+      const dependencies = nodesAvailable.filter((dependencyNode) => nodesId.includes(dependencyNode.id));
+      const minX = Math.min(currentNode.position.x, ...dependencies.map((value) => value.data.boundingBox!.topLeft.x));
+      const minY = Math.min(currentNode.position.y, ...dependencies.map((value) => value.data.boundingBox!.topLeft.y));
+      const maxX = Math.max(currentNode.position.x + nodeWidthClearance, ...dependencies.map((value) => value.data.boundingBox!.bottomRight.x));
+      const maxY = Math.max(currentNode.position.y + nodeHeightClearance, ...dependencies.map((value) => value.data.boundingBox!.bottomRight.y));
+      return {
+        topLeft: { x: minX, y: minY },
+        bottomRight: { x: maxX, y: maxY },
+      };
+    }
+    return {
+      topLeft: currentNode.position,
+      bottomRight: { x: currentNode.position.x + nodeWidthClearance, y: currentNode.position.y + nodeHeightClearance },
+    };
+  };
+
   /**
    * Calculate injects position when dragging stopped
    * @param nodeInjects the list of injects
    */
   const calculateInjectPosition = (nodeInjects: NodeInject[]) => {
-    nodeInjects.forEach((nodeInject, index) => {
-      let row = 0;
-      let rowFound = true;
+    let reorganizedInjects = nodeInjects;
+    for (let i = 0; i < nodeInjects.length; i += 1) {
+      let childrens = reorganizedInjects.slice(i).filter((nextNode) => nextNode.id !== nodeInjects[i].id
+          && nextNode.data.inject?.inject_depends_on !== undefined
+          && nextNode.data.inject?.inject_depends_on !== null
+          && nodeInjects[i].id in nextNode.data.inject!.inject_depends_on);
+
+      childrens = childrens.sort((a, b) => a.data.inject!.inject_depends_duration - b.data.inject!.inject_depends_duration);
+
+      for (let j = 0; j < childrens.length; j += 1) {
+        reorganizedInjects = moveItem(reorganizedInjects, i + j + 1, reorganizedInjects.indexOf(childrens[j], i));
+      }
+    }
```

Comment:
**nitpick (non-blocking):** it's possible to access the index with `forEach`

not-tested:
```
let reorganizedInjects = nodeInjects;

nodeInjects.forEach((node, i) => {
  let childrens = reorganizedInjects.slice(i).filter((nextNode) => 
    nextNode.id !== node.id &&
    nextNode.data.inject?.inject_depends_on !== undefined &&
    nextNode.data.inject?.inject_depends_on !== null &&
    node.id in nextNode.data.inject!.inject_depends_on
  );

  childrens = childrens.sort((a, b) => 
    a.data.inject!.inject_depends_duration - b.data.inject!.inject_depends_duration
  );

  childrens.forEach((child, j) => {
    reorganizedInjects = moveItem(reorganizedInjects, i + j + 1, reorganizedInjects.indexOf(child, i));
  });
});
```

Reply:
Yes, I'm always a bit reluctant to use functional when I know I might touch the array itself but in this case, I don't so you're right, that makes totally sense :) Doing !